src/datasets/arrow_dataset.py

```diff
@@ -2731,7 +2731,7 @@ def init_buffer_and_writer():
                 # Only load the columns we actually need
                 if input_columns:
                     input_dataset = self.with_format(
-                        self._format_type, columns=input_columns, output_all_columns=False, **self._format_kwargs
+                        self._format_type, columns=input_columns, output_all_columns=True, **self._format_kwargs
```

Comment:
If we do this I think we can remove this part in `apply_function_on_filtered_inputs` no ? Otherwise the data would be loaded twice if a format type is set

```python
            if self._format_type is not None:
                inputs = self._getitem(
                    key=(indices if isinstance(indices, int) else slice(indices[0], indices[-1] + 1)),
                    format_type=None,
                    format_columns=None,
                    format_kwargs=None,
                    decoded=False,
                )
```

Reply:
We still need this for the scenario where `input_columns` is `None` and self._format_type is not `None` (to ignore the formatting)

Reply:
Okay, now `output_all_columns` is still `False` if `input_columns` is specified, but we also execute the above block to add the non-`input_columns` .